canonical-web-and-design/maas-ui
# feat(ui) build fetched images table

## Done

- Built FetchedImages table which is used to sync images with a different source

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Go to /MAAS/r/images and click "Change source"
- Use a source type that's different to the current selection and click "Connect"
- Check that an image selection table shows
- Select an image and click "Update selection"
- It might take a while but eventually you should get taken back to the main image list, and the header should say "Syncing images from {the source URL you chose}"
- Wait until images start downloading and check that the "Change source" button becomes disabled
- Click "Stop import".
- It might take a while but eventually downloading will stop, then check that the "Change source" button becomes enabled again.

## Fixes

Fixes canonical-web-and-design/app-squad#141
